js/client.js

```diff
@@ -1325,8 +1325,8 @@
 				if (this.curSideRoom === oldRoom) this.curSideRoom = room;
 				if (this.sideRoom === oldRoom) this.sideRoom = room;
 			}
-			if (type === BattleRoom) this.roomList.push(room);
-			if (type === ChatRoom || type === BattlesRoom) this.sideRoomList.push(room);
+			if (type === BattleRoom && !room.isSideRoom) this.roomList.push(room);
+			if (type === ChatRoom || type === BattlesRoom || (type === BattleRoom && room.isSideRoom)) this.sideRoomList.push(room);
```

Comment:
I mean, test for teambuilder, ladder, home, and rooms, and don't push to roomlists at all.

Otherwise push depending on `isSideRoom`.